tests/iterator.rs

```diff
@@ -75,3 +91,122 @@ fn signals_block_wait() {
         .expect_err("Wait didn't wait properly");
     assert_eq!(err, RecvTimeoutError::Timeout);
 }
+
+#[test]
+#[serial]
+fn pending_doesnt_block_and_is_empty_if_no_signals_recieved() {
```

Comment:
nitpick: The name is quite a mouthful. I sometimes pick shorter names and put a doc comment onto the function (not that it would render in docs, but it has a feeling of being the documentation of the purpose of the thing as opposed to some random comment around it).